include/swift/AST/GenericSignatureBuilder.h

```diff
@@ -314,6 +314,22 @@ class GenericSignatureBuilder {
     /// \c ConstraintResult::Unresolved so the caller knows what happened.
     GenerateUnresolved = 1,
   };
+	
+  /// Describes a struct that contains information about the constraint
+  /// that didn't conform to a protocol or a class
+  struct InvalidConstraint {
+    Type subjectType;
+    Type constraintType;
+    const RequirementSource *reqSource;
+
+    InvalidConstraint(const Type type1, const Type type2,
+                      const RequirementSource *source)
+        : subjectType(type1), constraintType(type2), reqSource(source) {}
+  };
```

Comment:
This type is almost identical structurally to `Constraint<Type>`. Why not use `Constraint<Type>`?